### PR TITLE
fix: Rejoin headings with the subsequent first paragraph

### PR DESCRIPTION
**Pull request by yoomlam**
## Ticket

https://navalabs.atlassian.net/browse/DST-510

## Changes

Rejoin headings with the subsequent first paragraph so that headings are not cited by themselves.

## Testing

Updated unit test

### CODE REVIEW

File: app/src/citations.py [hunk 1]
```diff
@@ -26,7 +26,24 @@ def create_citation(self, chunk: Chunk, subsection: str) -> Subsection:
 
 
 def default_chunk_splitter(chunk: Chunk) -> list[str]:
```
Comment (yoomlam) [inline app/src/citations.py:28]: `default_chunk_splitter` is used by `split_into_subsections()` which is used in `BaseEngine`.

File: app/src/citations.py [hunk 2]
```diff
@@ -26,7 +26,24 @@ def create_citation(self, chunk: Chunk, subsection: str) -> Subsection:
...
+            skip_next = False
+            continue
+
+        next_split = splits[i + 1] if i + 1 < len(splits) else None
+        if next_split and split.startswith("#") and not next_split.startswith("#"):
+            better_splits.append(f"{split}\n{next_split}")
+            skip_next = True
+            continue
+
+        better_splits.append(split)
```
Comment (yoomlam) [inline app/src/citations.py:44]: This was originally an elegant one-liner but I expanded it to be more robust to account for headings without subsequent text -- see unit test. I'm open to code improvements to make it cleaner.

**Review by ccheng26 — APPROVED** [review]
lgtm, I'd prefer to leave `default_chunk_splitter` as is for readability

**Review by KevinJBoyer — APPROVED** [review]
Forgot to add my review, LGTM!

**Pull request merged**